### PR TITLE
[Feature][Doc] imporve doc about connector-v2 schema feature

### PR DESCRIPTION
**Pull request by liugddx**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by liugddx** [comment]
@EricJoy2048 @Hisoka-X PTAL

**Review by Hisoka-X — DISMISSED** [review]

**Comment by Hisoka-X** [comment]
@TyrantLucifer PTAL.

**Review by EricJoy2048 — DISMISSED** [review]

**Comment by EricJoy2048** [comment]
@TyrantLucifer  PTAL.

File: docs/en/concept/schema-feature.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ Some NoSQL databases or message queue are not strongly limited schema, so the sc
 | time      | Only store time. Accuracy is 100 nanoseconds.                |
 | timestamp | Stores a unique number that is updated whenever a row is created or modified. timestamp is based on the internal clock and does not correspond to real time. There can only be one timestamp variable per table. |
 | row       | Row type,can be nested.                                      |
+| map       | A Map is an object that maps keys to values.                 |
```
Comment (TyrantLucifer) [inline docs/en/concept/schema-feature.md]: Please add the type restriction case for array and map.

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: docs/en/concept/schema-feature.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ Some NoSQL databases or message queue are not strongly limited schema, so the sc
...
+| array     | A array is a data type that represents a collection of elements                                  |
 
 ## How to use schema
```
Comment (TyrantLucifer) [inline docs/en/concept/schema-feature.md:29]: Please also add array and map examples in there.

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

**Review by TyrantLucifer — APPROVED** [review]
+1, CC @Hisoka-X @EricJoy2048

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**